build-system/pr-check.js

```diff
@@ -257,27 +259,25 @@ const command = {
 
 function runAllCommands() {
   // Run different sets of independent tasks in parallel to reduce build time.
-  if (process.env.BUILD_SHARD == "pre_build_checks") {
+  if (process.env.BUILD_SHARD == "pre_build_checks_and_unit_tests") {
     command.testBuildSystem();
-    command.runPreBuildChecks();
+    command.cleanBuild();
+    command.buildRuntime();
+    command.runLintChecks();
     command.runDepAndTypeChecks();
-    // Skip testDocumentLinks() during push builds.
+    command.runUnitTests();
+    // command.testDocumentLinks() is skipped during push builds.
     command.buildValidatorWebUI();
     command.buildValidator();
   }
   if (process.env.BUILD_SHARD == "integration_tests") {
+    command.cleanBuild();
     command.buildRuntime();
-    command.presubmit();  // Must be run after the runtime is built.
+    command.serveRuntime();
+    command.runPresubmitTests();  // Needs runtime to be built and served.
```

Comment:
My understanding is that `gulp presubmit` can scan through the generated files in dist/ for errors, and that's why it's good to run presubmit after dist.

The current state is that we're running `gulp presubmit` pre- and post- submit. The difference though, is that since we don't do a dist for PRs, the directories in dist/ will only be covered on master. We will cover the directories in `src/` for all PRs.

I ran this by @erwinmombay, who seemed to be fine with this.